bundles/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/runnable/WizLightingUpdateReceiverRunnable.java

```diff
@@ -0,0 +1,146 @@
+
+    private boolean shutdown;
+    private int listeningPort;
+
+    /**
+     * Constructor of the receiver runnable thread.
+     *
+     * @param mediator the {@link WizLightingMediator}
+     * @param listeningPort the listening UDP port
+     * @throws SocketException is some problem occurs opening the socket.
+     */
+    public WizLightingUpdateReceiverRunnable(final WizLightingMediator mediator, final int listeningPort)
+            throws SocketException {
+        logger.debug("Starting Update Receiver Runnable...");
+        this.listeningPort = listeningPort;
+        this.mediator = mediator;
+
+        // Create a socket to listen on the port.
+        logger.debug("Opening socket and start listening UDP port: {}", listeningPort);
+        DatagramSocket dsocket = new DatagramSocket(null);
+        dsocket.setReuseAddress(true);
+        dsocket.setBroadcast(true);
+        dsocket.setSoTimeout(TIMEOUT_TO_DATAGRAM_RECEPTION_MILLISECONDS);
+        dsocket.bind(new InetSocketAddress(listeningPort));
+        this.datagramSocket = dsocket;
+        logger.debug("Update Receiver Runnable and socket started with success...");
+
+        this.shutdown = false;
+    }
+
+    @Override
+    public void run() {
+        // Now loop forever, waiting to receive packets and redirect them to mediator.
+        while (!this.shutdown) {
+            datagramSocketHealthRoutine();
+
+            // Create a buffer to read datagrams into. If a
+            // packet is larger than this buffer, the
+            // excess will simply be discarded!
+            byte[] buffer = new byte[2048];
+
+            // Create a packet to receive data into the buffer
+            DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+
+            // Wait to receive a datagram
+            try {
+                datagramSocket.receive(packet);
+
+                logger.trace("Runnable received packet from: {}. Will process the packet...",
+                        packet.getAddress().getHostAddress());
+
+                // Redirect packet to the mediator
+                WizLightingResponse response = this.packetConverter.transformResponsePacket(packet);
+                if (response != null) {
+                    this.mediator.processReceivedPacket(response);
+                } else {
+                    logger.debug("No WizLightingResponse was parsed from returned packet");
+                }
+            } catch (SocketTimeoutException e) {
+                logger.trace("No incoming data on port {} during {} ms socket was listening.", listeningPort,
+                        TIMEOUT_TO_DATAGRAM_RECEPTION_MILLISECONDS);
+            } catch (IOException e) {
+                logger.debug("One exception has occurred: {} ", e.getMessage());
+            }
+        }
+
+        // close the socket
+        logger.trace("Ending run loop; closing socket.");
+        datagramSocket.close();
```

Comment:
This entire run method should be wrapped in a try/finally so that you can make sure that this socket gets closed.

Reply:
Done.